app/src/main/java/com/github/se/orator/model/symblAi/AudioRecorder.kt

```diff
@@ -47,7 +51,10 @@ class AudioRecorder(
         AudioRecord(
             MediaRecorder.AudioSource.MIC, sampleRate, channelConfig, audioFormat, bufferSize)
 
-    audioFile = File(context.cacheDir, "audio_record.wav") // Recording in WAV format
+    // Save file to the appropriate directory based on offline mode
+    val saveDir = if (isOffline) context.filesDir else context.cacheDir
+    audioFile =
+        File(saveDir, "audio_record_${System.currentTimeMillis()}.wav") // Recording in WAV format
```

Comment:
After some research, I found that the phone cache can be automatically deleted in case the phone is full or in case of problems. 
Grosso modo, filesDir is the file directory of the permanent files. There are more information on this [page](https://developer.android.com/training/data-storage/app-specific) if you want.

Reply:
Thanks !